PocketKit/Sources/PocketKit/Home/HomeCarouselItemCell.swift

```diff
@@ -0,0 +1,285 @@
+        contentView.layoutMargins = Constants.layoutMargins
+
+        mainContentStack.translatesAutoresizingMaskIntoConstraints = false
+        thumbnailView.translatesAutoresizingMaskIntoConstraints = false
+        bottomStack.translatesAutoresizingMaskIntoConstraints = false
+
+        thumbnailWidthConstraint = thumbnailView.widthAnchor.constraint(
+            equalToConstant: Constants.thumbnailSize.width
+        ).with(priority: .required)
+
+        contentView.layoutMargins = Constants.layoutMargins
+        NSLayoutConstraint.activate([
+            mainContentStack.topAnchor.constraint(equalTo: contentView.layoutMarginsGuide.topAnchor),
+            mainContentStack.leadingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leadingAnchor),
+            mainContentStack.trailingAnchor.constraint(equalTo: contentView.layoutMarginsGuide.trailingAnchor),
+
+            thumbnailView.heightAnchor.constraint(equalToConstant: Constants.thumbnailSize.height).with(priority: .required),
+            thumbnailWidthConstraint!,
+
+            bottomStack.leadingAnchor.constraint(equalTo: mainContentStack.leadingAnchor),
+            bottomStack.trailingAnchor.constraint(equalTo: mainContentStack.trailingAnchor),
+            bottomStack.bottomAnchor.constraint(equalTo:  contentView.layoutMarginsGuide.bottomAnchor).with(priority: .required),
+        ])
+        
+        [UIView(), domainLabel, timeToReadLabel, UIView()].forEach(subtitleStack.addArrangedSubview)
+        [favoriteButton, saveButton, overflowButton].forEach(buttonStack.addArrangedSubview)
+        [titleLabel, thumbnailView].forEach(mainContentStack.addArrangedSubview)
+        [subtitleStack, UIView(), buttonStack].forEach(bottomStack.addArrangedSubview)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension HomeCarouselItemCell {
+    struct Model: Hashable {
+        let title: String?
+        let domain: String?
+        let timeToRead: String?
+        let thumbnailURL: URL?
+        let saveButtonMode: RecommendationSaveButton.Mode?
+        
+        init(item: ItemsListItem) {
+            self.title = item.title
+            self.domain = item.domainMetadata?.name ?? item.domain ?? item.bestURL?.host
+            let timeToRead = item.timeToRead ?? 0
+            self.timeToRead = timeToRead > 0 ? "\(timeToRead) min read" : nil
+            self.thumbnailURL = imageCacheURL(for: item.topImageURL)
+            self.saveButtonMode = nil
+        }
+        
+        init(viewModel: HomeRecommendationCellViewModel) {
+            self.title = viewModel.title
+            self.domain = viewModel.domain
+            self.timeToRead = viewModel.timeToRead
+            self.thumbnailURL = viewModel.imageURL
+            self.saveButtonMode = viewModel.saveButtonMode
+        }
+
+        var favoriteAction: ItemAction? = nil
+        var overflowActions: [ItemAction]? = nil
```

Comment:
I found myself looking above the initializer for these properties since they're not computed. Would you mind moving them?

Generally speaking, here's how we like to approach ordering property and function declarations:
```swift
class Foo {
    // declare public properties at the top
    var bar: Bar?
    let baz: Baz

    // private properties come next
    private let qux: Qux

    // then comes any initializers (public first, then private)
    public init() {
        self.qux = Qux.shared
    }
    
    private init(qux: Qux) {
        self.qux = qux
    }

    // method declarations come after the initializer
    func quux() {
    }

    // also, complex computed properties can be placed along with public functions
    // since, practically speaking, they are just parameterless functions
    var quuz: Quuz {
        if foo != nil {
            foo.bar().map { $0.otherThing }
        } else {
            []
        }
    }
}

// also can be nice to group like functions in an extension w/ a `MARK: - Group Title` comment
// MARK: - Network calls
extension Foo {

    func fetchSomething() async {
        // network related code
    }
}
```

There is probably some debate between @dskuza and I regarding placing computed properties above or below the initializer, so this isn't set in stone. Also there are reasons to deviate from this convention (like if a property is closely related to a single function) but this is our general approach.